lib/inferno/repositories/results.rb

```diff
@@ -15,8 +15,18 @@ def create(params)
         messages = params.delete(:messages) || []
         requests = params.delete(:requests) || []
         super(params).tap do |result|
+          result_model = self.class::Model.find(id: result.id)
           messages.each { |message| messages_repo.create(message.merge(result_id: result.id)) }
-          requests.each { |request| requests_repo.create(request.to_hash.merge(result_id: result.id)) }
+          requests.each do |request|
+            request_created =
+              if request.result_id.nil?
+                requests_repo.create(request.to_hash.merge(result_id: result.id))
+              else
+                requests_repo.create(request.to_hash)
```

Comment:
Am I wrong to expect that if it already has a `result_id`, then it doesn't need to be created because this is a used request rather than a made request? If it already has an `id`, that might be a more straightforward way to determine whether it's already been persisted.

I'd expect something like this so that we aren't creating duplicate entries in the requests table:
```
request_id =
  if request.id.present?
    request.id
  else
    requests_repo.create(request.to_hash.merge(result_id: result.id)).id
  end
```